### PR TITLE
Multi download operation

### LINKED ISSUE #1078: expose Multi download operation
expose multiDownload in all these 3 places

feat wasm: expose MultiDownload
feat mobile: expose MultiDownload
feat winsdk: expose MultiDownload

### PR DESCRIPTION
**Pull request by Hitenjain14**
### Changes
- feat wasm: expose `MultiDownload` 
- feat mobile: expose `MultiDownload`
- feat winsdk: expose `MultiDownload`

### Fixes
- https://github.com/0chain/gosdk/issues/1078

### Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/gosdk/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

### Associated PRs (Link as appropriate):
- blobber:
- 0chain:
- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

### CODE REVIEW

**Comment by boddumanohar** [comment]
@Hitenjain14  please also replace the current [current download method in wasm demo](https://github.com/0chain/gosdk/blob/1df91bf714ae22957348a532998263519585ecda/wasmsdk/demo/index.html#L470) with multiDownload

File: mobilesdk/zbox/storage.go [hunk 1]
```diff
@@ -250,6 +258,46 @@ func DownloadThumbnail(allocationID, remotePath, localPath string, statusCb Stat
...
+		if options[i].DownloadOp == 1 {
+			err = a.DownloadFile(options[i].LocalPath, options[i].RemotePath, false, &StatusCallbackWrapped{Callback: statusCb}, false)
+		} else {
+			err = a.DownloadThumbnail(options[i].LocalPath, options[i].RemotePath, false, &StatusCallbackWrapped{Callback: statusCb}, false)
+		}
+		if err != nil {
+			return err
+		}
+	}
+	if options[len(options)-1].DownloadOp == 1 {
```
Comment (dabasov) [inline mobilesdk/zbox/storage.go]: can there be 0 options? please add a check

File: zboxcore/sdk/read_counter_ctr.go [hunk 2]
```diff
@@ -5,13 +5,36 @@ import (
...
+	}
+	mut := brc.blobberLockMap[key]
+	brc.muBlobberLock.Unlock()
+	mut.Lock()
+}
+
+func unlockBlobberReadCtr(allocID, blobberID string) {
+	brc.muBlobberLock.Lock()
+	key := allocID + blobberID
+	mut := brc.blobberLockMap[key]
```
Comment (dabasov) [inline zboxcore/sdk/read_counter_ctr.go:35]: check if nothing is found din the map

File: zboxcore/sdk/allocation.go [hunk 3]
```diff
@@ -292,25 +291,15 @@ func (a *Allocation) startWorker(ctx context.Context) {
...
 
 func (a *Allocation) dispatchWork(ctx context.Context) {
-	sem := semaphore.NewWeighted(processDownloadWorkerCount)
 	for {
 		select {
 		case <-ctx.Done():
 			l.Logger.Info("Upload cancelled by the parent")
 			return
 		case downloadReq := <-a.downloadChan:
-			err := sem.Acquire(ctx, 1)
```
Comment (dabasov) [inline zboxcore/sdk/allocation.go:302]: why do we create goroutines without any control?

**Review by dabasov — CHANGES_REQUESTED** [review]

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | false  |
| system-tests  | master  |
| gosdk  | feature/mutli-download  |
| zboxcli  | sprint-july-1  |
| zwalletcli  | sprint-july-1  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0dns  | staging  |

**Review by dabasov — APPROVED** [review]
lgtm

**Pull request merged**